src/librustc_mir/interpret/intrinsics.rs

```diff
@@ -239,7 +239,25 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
             "transmute" => {
                 self.copy_op_transmute(args[0], dest)?;
             }
-
+            "simd_insert" => {
+                let mut vector = self.read_vector(args[0])?;
+                let index = self.read_scalar(args[1])?.to_u32()? as usize;
+                let scalar = self.read_immediate(args[2])?;
+                if vector[index].layout.size == scalar.layout.size {
+                    vector[index] = scalar;
+                } else {
+                    throw_ub_format!(
```

Comment:
this is only reachable if the intrinsic is used wrongly, which any stabilized wrapper should never do, right? If so, just use an assert, there's no need to fail nicely on mis-use of an intrinsic

Reply:
> this is only reachable if the intrinsic is used wrongly, which any stabilized wrapper should never do, right? 

Right. 

> If so, just use an assert, there's no need to fail nicely on mis-use of an intrinsic

Will do.